### PR TITLE
[DOCS] Jailbreak Snapshot and Restore API documentation to REST API section

### PR DESCRIPTION
**Pull request by wchrisdean**

### CODE REVIEW

File: docs/reference/snapshot-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,184 @@
+[[snapshot-restore-api]]
+=== Snapshot Restore API
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Titles and headings should be sentence case.
```suggestion
=== Snapshot restore API
```

File: docs/reference/snapshot-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+++++
+<titleabbrev>Snapshot Restore API</titleabbrev>
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
<titleabbrev>Snapshot restore</titleabbrev>
```

File: docs/reference/snapshot-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+++++
+////
+Restores a snapshot into a running cluster.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: These comments seem redundant. I'd remove them.

File: docs/reference/snapshot-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+////
+***********************************************************
+////
+
+////
+Use the appropriate heading levels for your book.
+Add anchors for each section.
+FYI: The section titles use attributes in case those terms change.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove

File: docs/reference/snapshot-api.asciidoc [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+[[sample-api-request]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: This looks like the placeholder ID from the template. This needs to be unique to the API. Here's a suggestion:

```suggestion
[[snapshot-restore-api-request]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+==== {api-request-title}
+////
+This section show the basic endpoint, without the body or optional parameters.
+Variables should use <...> syntax.
+If an API supports both PUT and POST, include both here.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove

File: docs/reference/snapshot-api.asciidoc [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+==== {api-prereq-title}
+////
+Optional list of prerequisites.
+
+For example:
+
+* A snapshot of an index created in 5.x can be restored to 6.x. You must...
+* If the {es} {security-features} are enabled, you must have `write`, `monitor`,
+and `manage_follow_index` index privileges...
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: You can omit this section is it's unneeded.

File: docs/reference/snapshot-api.asciidoc [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+This section show the basic endpoint, without the body or optional parameters.
+Variables should use <...> syntax.
+If an API supports both PUT and POST, include both here.
+////
+
+`PUT /<follower_index>/_ccr/follow`
+
+
+[[sample-api-prereqs]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-prereqs]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+For example:
+
+* A snapshot of an index created in 5.x can be restored to 6.x. You must...
+* If the {es} {security-features} are enabled, you must have `write`, `monitor`,
+and `manage_follow_index` index privileges...
+////
+
+
+[[sample-api-desc]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-desc]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+* Use a definition list.
+* End each definition with a period.
+* Include whether the parameter is Optional or Required and the data type.
+* Include default values as the last sentence of the first paragraph.
+* Include a range of valid values, if applicable.
+* If the parameter requires a specific delimiter for multiple values, say so.
+* If the parameter supports wildcards, ditto.
+* For large or nested objects, consider linking to a separate definition list.
+***************************************
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Omit the comments and replace with a longer form description of the API and it's intended usage.

File: docs/reference/snapshot-api.asciidoc [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-path-params]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-path-params]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+==== {api-path-parms-title}
+////
+A list of all the parameters within the path of the endpoint (before the query string (?)).
+
+For example:
+`<follower_index>`::
+(Required, string) Name of the follower index
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove these comments. This is where you define the variable parameters you set up in the request section.

File: docs/reference/snapshot-api.asciidoc [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-query-params]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-query-params]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+A list of the parameters in the query string of the endpoint (after the ?).
+
+For example:
+`wait_for_active_shards`::
+(Optional, integer) Specifies the number of shards to wait on being active before
+responding. A shard must be restored from the leader index being active.
+Restoring a follower shard requires transferring all the remote Lucene segment
+files to the follower index. The default is `0`, which means waiting on none of
+the shards to be active.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove

File: docs/reference/snapshot-api.asciidoc [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-request-body]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-request-body]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+A list of the properties you can specify in the body of the request.
+
+For example:
+`remote_cluster`::
+(Required, string) The <<modules-remote-clusters,remote cluster>> that contains
+the leader index.
+
+`leader_index`::
+(Required, string) The name of the index in the leader cluster to follow.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove

File: docs/reference/snapshot-api.asciidoc [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-response-body]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-response-body]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+`auto_follow_stats`::
+  (object) An object representing stats for the auto-follow coordinator. This
+  object consists of the following fields:
+
+`auto_follow_stats.number_of_successful_follow_indices`:::
+  (long) the number of indices that the auto-follow coordinator successfully
+  followed
+...
+
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove.

File: docs/reference/snapshot-api.asciidoc [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-response-codes]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-response-codes]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+////
+Response codes are only required when needed to understand the response body.
+
+For example:
+`200`::
+Indicates all listed indices or index aliases exist.
+
+ `404`::
+Indicates one or more listed indices or index aliases **do not** exist.
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove

File: docs/reference/snapshot-api.asciidoc [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+
+
+[[sample-api-example]]
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]:
```suggestion
[[snapshot-restore-api-example]]
```

File: docs/reference/snapshot-api.asciidoc [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+[source,js]
+----
+{
+  "follow_index_created" : true,
+  "follow_index_shards_acked" : true,
+  "index_following_started" : true
+}
+----
+// TESTRESPONSE
+////
```
Comment (jrodewig) [inline docs/reference/snapshot-api.asciidoc]: Remove.

**Review by debadair — CHANGES_REQUESTED** [review]
Agree with @jrodewig's comments. In general, I think we should be consistent and name the source files the same as the anchors that are used as the generated HTML file names. It's easier all the way around if there's a 1:1 mapping between source files and the generated HTML files. 

I'd actually consider splitting Snapshot and Restore into two separate reference pages. While there's a strong argument for keeping them together from the API view, from the user perspective they are two separate actions and having them separate would be better from a discoverability standpoint. If I didn't already know, I might see "Snapshot restore" as the API to restore a snapshot and wonder where "Snapshot create" was.

**Pull request closed without merging**